docs/design/storage-node-deployment-windows-home.md

```diff
@@ -0,0 +1,146 @@
+* Error gui application
+    * shows errors from log file
+    * notifies user of service errors.
+    * saves last reported error timestamp to a file for knowing if there are unread errors.
+
+### Automatic Updates
+
+Finding out the minimum version and latest stable.
+
+General
+* Windows firewall can block storage node operations.
+    * [isportallowed](https://docs.microsoft.com/en-us/windows/win32/api/netfw/nf-netfw-inetfwmgr-isportallowed) windows api function can be used to make sure we are allowed through firewall
+    * Can we add code to detect if firewall is blocking storage node operations?
+    * Unblock storage node operator in firewall settings.
+    * Can we detect if windows firewall is running?
+
+Download a new binary.
+* download fails
+    * Log error in log file if possible
+    * Retry download on next cycle that checks if storage node is up to date.
+* out-of-space for downloading
+    * Log error in log file if possible
+    * Retry download on next cycle that checks if storage node is up to date.
+* filesystem read-only
+    * Log error in log file if possible
+    * Retry download on next cycle that checks if storage node is up to date.
+* MITM attacks/corrupted binary
+    * Log error in log file if possible
+    * Verify binary hashes with message from version server and with output of a hashing algorithm (shasum256)
+    * Retry download on next cycle that checks if storage node is up to date.
+
+Swapping in a new binary. 
+* computer crashes during swapping
+    * Log error in log file if possible
+    * automatic updater checks binary version and reruns download/swap steps.
+* deletion/stopping of the old binary fails.
+    * Log error in log file if possible
+    
+Starting a new binary.
+What if:
+* out-of-space during migrations,
+    * Log error in log file if possible
+    * automatic updater will try to rerun the binary on next cycle
+* failure to start.
+    * Log error in log file if possible
+    * automatic updater will try to rerun the binary on next cycle
+* not yet configured.
+    * Log error in log file if possible
+    * storage node will run a setup or describe out to fix the problem...?
+
+Gradual Rollouts.
+* bad gradual rollout. We know it's a bad rollout if our application stops working
+    * Log error in log file if possible
+    * if we did have a database migration, api/grpc change, or file system change in the latest update then wait for next update???
+        * bad latest update might have had faulty database changes that will need to be migrated again.
+    * otherwise rollback to previous version
+
+### Starting Storage Node binary on OS Start-Up
+
+We need to ensure that updater binary starts on computer start-up,
+without logging into the system, and this updater binary launches the storage node This is achieved when installing via the msi.
+
+### Restarting Storage Node binary on Crash / Problems
+
+We need to ensure that storage node binary restarts after a crash.
```

Comment:
A windows service can be configured to restart the service on fail/crash same can do an upstart script under linux.